### PR TITLE
feat(infra): adding add-to-project to projen

### PR DESCRIPTION
**Pull request by rudrateja-clickup**
Fixes #

This PR adds add-to-project to projen. 

Associated Task: https://staging.clickup.com/t/333/CLK-228826

### CODE REVIEW

File: .github/workflows/renovate.yml [hunk 1]
```diff
@@ -39,3 +39,16 @@ jobs:
           RENOVATE_ENABLED_MANAGERS: '["npm"]'
           RENOVATE_REPOSITORIES: '["${{ github.repository }}"]'
           RENOVATE_FORCE: ${{ github.event_name == 'workflow_dispatch' && '{"schedule":null}' || '' }}
+  add-to-project:
```
Comment (mattlewis92) [inline .github/workflows/renovate.yml:42]: I think this needs to be in its own workflow file that runs:

```
on:
  pull_request:
    types:
      - opened
```

File: .github/workflows/add-to-project.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# ~~ Generated by projen. To modify, edit .projenrc.js and run "npx projen".
+
+name: add-to-project
+on:
+  pull_request:
+    types:
+      - opened
+      - labeled
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'workflow_dispatch' && github.event_name || '' }}
```
Comment (mattlewis92) [inline .github/workflows/add-to-project.yml:10]: I think we can just make this `${{ github.workflow }}-${{ github.ref_name }}`

File: .github/workflows/add-to-project.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    permissions:
+      contents: read
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Add issue to project
+        uses: actions/add-to-project@v0.4.1
+        with:
+          project-url: https://github.com/orgs/time-loop/projects/3
+          github-token: x-access-token:${{ secrets.PROJEN_GITHUB_TOKEN }}
```
Comment (mattlewis92) [inline .github/workflows/add-to-project.yml:23]: I think this should be `github-token: ${{ secrets.PROJEN_GITHUB_TOKEN }}`

File: .github/workflows/add-to-project.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+      - labeled
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'workflow_dispatch' && github.event_name || '' }}
+jobs:
+  add-to-project:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - name: Checkout
```
Comment (mattlewis92) [inline .github/workflows/add-to-project.yml:17]: I don't think we actually need the checkout step here?

File: src/add-to-project.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  export const GITHUB_PROJECT_NUMBER = '3';
+
+  const defaultWorkflow = {
+    name: 'add-to-project',
+    on: {
+      pull_request: {
+        types: ['opened', 'labeled'],
+      },
+    },
+    // Prevent renovate running multiple times in parallel due to several subsequent trigger events occurring
```
Comment (mattlewis92) [inline src/add-to-project.ts:16]: We can remove this comment as well when we change the concurrency grouping

Comment (jeisson-clickup) [reply src/add-to-project.ts]: Updated [33832a8](https://github.com/time-loop/clickup-projen/pull/129/commits/33832a87511692f6c8cc2a8c80e5417a339543f1)

File: .github/workflows/add-to-project.yml (hunk 4 above)
Comment (jeisson-clickup) [reply .github/workflows/add-to-project.yml]: Updated [33832a8](https://github.com/time-loop/clickup-projen/pull/129/commits/33832a87511692f6c8cc2a8c80e5417a339543f1)

File: .github/workflows/add-to-project.yml (hunk 3 above)
Comment (jeisson-clickup) [reply .github/workflows/add-to-project.yml]: Updated [33832a8](https://github.com/time-loop/clickup-projen/pull/129/commits/33832a87511692f6c8cc2a8c80e5417a339543f1)

File: .github/workflows/add-to-project.yml (hunk 2 above)
Comment (jeisson-clickup) [reply .github/workflows/add-to-project.yml]: Updated [33832a8](https://github.com/time-loop/clickup-projen/pull/129/commits/33832a87511692f6c8cc2a8c80e5417a339543f1)

File: .github/workflows/renovate.yml (hunk 1 above)
Comment (jeisson-clickup) [reply .github/workflows/renovate.yml]: updated [feb3c8a](https://github.com/time-loop/clickup-projen/pull/129/commits/feb3c8a6c249a0a7f03664cbf8b4e193a50858ce)

**Review by jeisson-clickup — APPROVED** [review]

**Pull request merged**